python/lsst/obs/base/gen2to3/convert_tests.py

```diff
@@ -0,0 +1,208 @@
+        # TODO: remove the path.join and getpackagedir once I'm done developing the bin.src version:
+        self.cmd = os.path.join(lsst.utils.getPackageDir("obs_base"), "bin.src/convert_gen2_repo_to_gen3.py")
+        self.collections.add(self.instrumentName)
+        if len(self.refcats) > 0:
+            self.collections.add("refcats")
+
+    def tearDown(self):
+        shutil.rmtree(self.gen3root, ignore_errors=True)
+
+    def _run_convert(self):
+        """Convert a gen2 repo to gen3 for testing.
+        """
+        cmd = [self.cmd, self.instrumentClass,
+               "--gen2root", self.gen2root,
+               "--gen3root", self.gen3root,
+               "--calibs", self.gen2calib
+               ]
+        if self.config is not None:
+            cmd.extend(("--config", self.config))
+        if self.args is not None:
+            cmd.extend(self.args)
+        print(f"Running command: {' '.join(cmd)}")
+        subprocess.run(cmd)
+
+    def check_raw(self, gen3Butler, exposure, detector):
+        """Check that a raw was converted correctly."""
+        dataIdGen2 = dict(ccd=detector, visit=exposure)
+        try:
+            gen2Exposure = self.gen2Butler.get('raw', dataId=dataIdGen2)
+        except lsst.daf.persistence.butlerExceptions.NoResults:
+            # ignore datasets that don't actually exist in the gen2 butler.
+            return
+        dataIdGen3 = dict(detector=detector, exposure=exposure, instrument=self.instrumentName)
+        gen3Exposure = gen3Butler.get('raw', dataId=dataIdGen3)
+        # Check that we got an Exposure, but not what type; there is
+        # inconsistency between different obs packages.
+        self.assertIsInstance(gen3Exposure, lsst.afw.image.Exposure)
+        self.assertEqual(gen3Exposure.getInfo().getDetector().getId(), detector)
+        self.assertMaskedImagesEqual(gen2Exposure.maskedImage, gen3Exposure.maskedImage)
+
+    def check_calibs(self, calibName, calibIds, gen3Butler):
+        """Test that we can get converted bias/dark/flat from the gen3 repo.
+
+        Note: because there is no clear way to get calibrations from a gen2
+        repo, we just test that the thing we got is a ExposureF here, and
+        assume that formatter testing is handled properly elsewhere.
+
+        Parameters
+        ----------
+        calibName : `str`
+            The name of the calibration to attempt to get ('bias', 'flat').
+        calibIds : `list` of `dict`
+            The list of calibration dataIds to get.
+        gen3Butler : `lsst.daf.butler.Butler`
+            The Butler to use to get the data.
+        """
+        for dataId in calibIds:
+            gen3Exposure = gen3Butler.get(calibName, dataId=dataId)
+            # TODO: how to test that we got the right detector?
+            # For DECam, this works and is a good check that we used the right Formatter.
+            # For HSC, the Exposure doesn't have an associated `Detector` object. Which is probably bad?
```

Comment:
Why we don't just write those to the master calibration metadata directly (instead of packign them into a string), I wish I knew.

Since the one that really matters here is decam and we probably won't be able to test it as part of CI anyway (see my comments on the obs_decam PR), I'm just going to skip this.